Automattic/sensei
# Improve support for PHP 8.1

Resolves #7177

This PR improves the PHP 8.1 support by following the WP.com PHP 8.1 migration guidelines (p5TWut-OG-p2). 

## Proposed Changes

* Improve PHP 8.1 support by applying fixes generated by [wpcom-php-migration](https://github.com/Automattic/wpcom-php-migration). I think some of the Rector suggestions were false positive, so I've skipped those. 
* Fix an 8.1 deprecation warning that happens when sending emails.

## Testing Instructions
1. Set up the plugin on PHP 8.1.
2. Test the main features and make sure there are no errors.
3. (optional) Run the [wpcom-php-migration](https://github.com/Automattic/wpcom-php-migration) on the repository and check if something is a miss.
  
## Pre-Merge Checklist
